src/lib.rs

```diff
@@ -93,6 +93,30 @@ pub struct Config {
     check_file_created: bool,
 }
 
+/// Represents the types of errors that may occur while using gcc-rs.
+#[derive(Clone, Debug)]
+pub enum ErrorKind {
+    /// Error occurred while performing I/O.
+    IOError,
+    /// Invalid architecture supplied.
+    ArchitectureInvalid,
+    /// Environment variable not found, with the var in question as extra info.
+    EnvVarNotFound,
+    /// Error occurred while using external tools (ie: invocation of compiler).
+    ToolExecError,
+    /// Error occurred due to missing external tools.
+    ToolNotFound,
+}
+
+/// Represents an internal error that occurred, with an explaination.
+#[derive(Clone, Debug)]
+pub struct Error {
+    /// Describes the kind of error that occurred.
+    pub kind: ErrorKind,
```

Comment:
We typically try to keep the fields of a struct private for maximal forwards compatibility, so could the `ErrorKind` enum and the fields of this struct be left private? We could always publicly expose them later, however!